sed/binning.py

```diff
@@ -95,6 +96,45 @@ def _simplify_binning_arguments(
     return bins, axes, ranges
 
 
+def bin_edges_to_bin_centers(bin_edges: np.array) -> np.array:
+    """Converts a list of bin edges into corresponding bin centers
+
+    Args:
+        bin_edges: 1d array of bin edges
+
+    Returns:
+        bin_centers: 1d array of bin centers
+    """
+
+    bin_centers = bin_edges[:-1] + np.diff(bin_edges) / 2
```

Comment:
The purpose is to use these functions to convert input bin centers to bin edges for the binning. This is nontrivial when the bin centers are non-uniform, which is the only reason to use input bins as a list.

Reply:
My point was that a function should encapsulate a "complex" operation. IMHO, a sum is not complex enough to deserve a function by itself because, at the end of the day, a function call adds a bit of overhead.

Reply:
In my opinion, these functions are very useful for inexperienced users, which just want to "convert the bins" without thinking about how to mathematically approach the problem, nor learn about all `numpy` functions.

We need to keep in mind that the main task of this package is to be user friendly. I don't believe that the overheads of a function call are ever going to impact the performance to a significant level, especially for functions you call once.

Reply:
That function will be called from another "function".
Neither the user nor you will have to call it directly ever, so I don't see the need for it as it's just doing "a simple operation". That was point.

To be honest, it's not something I'd write but I don't have a strong opinion about keeping it, if all others seems to be happy about it.

Reply:
I see your point, and could in general agree. However in this case, this function will be useful also outside this function which calls it, and having it accessible to the users is going to be helpful.

Reply:
And just to reinforced my point, the numbers in the following benchmark is crystal clear:
![image](https://user-images.githubusercontent.com/4411314/185384897-561dd812-31be-4ac3-835b-4c11c77f6789.png)

Reply:
You are comparing pears an oranges. The two functions don't compute the same if the input is not equidistant!

Reply:
@That's not true in this particular case. The input/output is exactly the same for both functions.

Could you give another input, so I could make my mind clearly into the problem? I can't help you with the code if I don't understand the problem ;)

Reply:
Ahh, not I see what you are doing. Okay, I can adopt that. It's more difficult to understand (for me), but I can add a respective comment line. Anyways, these are very small fish to fry, and not much worth the discussion. This is called once. Let's concentrate on adding functionality, or improving efficiency on the big scale (without changing functionality).